acme/acme/client_test.py

```diff
@@ -127,6 +127,13 @@ def test_query_registration(self):
         self.response.json.return_value = self.regr.body.to_json()
         self.assertEqual(self.regr, self.client.query_registration(self.regr))
 
+    def test_query_registration_updates_new_authzr_uri(self):
```

Comment:
Ah, I was wrong, indeed - that test wouldn't pass in the previous version of the code because `_send_recv_regr` takes data from supplied `regr`.

Resolved.